### PR TITLE
BUILD(macOS): mitigate build issues with OpenSSL 3

### PR DESCRIPTION
**Pull request by saerdnaer**
c.f. 
* https://github.com/mumble-voip/mumble/issues/5277#issuecomment-939513021
* https://github.com/signalwire/libks/commit/3cc75aba7875d3a32c7edaf64a1484c011a37955

### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

### CODE REVIEW

**Comment by Krzmbrzl** [comment]
Okay, so I take it, this now allows you to successfully build Mumble with OpenSSL 3, right? However, I don't think we should instruct people to disable warnings-as-errors from the beginning, since that will probably result in warnings being ignored.

The proper fix for the issue would of course be to properly migrate to OpenSSL 3, but for the time being I'd be fine with your code changes. I think you should drop the change to the build instructions though.

And for your commit message: I think `Mitigate` is a better description for what has been done, than `"fix"` :)

**Comment by Krzmbrzl** [comment]
According to https://github.com/mumble-voip/mumble/issues/4266#issuecomment-969672537 the memcheck block should be removed entirely

**Comment by saerdnaer** [comment]
@Krzmbrzl Should I update the PR - aka fix the commit message and remove the code block?

**Comment by Krzmbrzl** [comment]
We don't squash on merge - I think that was because that would remove the merge commit itself (which we use for generating our changelog)

**Comment by saerdnaer** [comment]
@Krzmbrzl squashed commits into one

File: docs/dev/build-instructions/build_macos.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ It is recommended to perform a so-called "out-of-source-build". In order to do s
 following commands:
 1. `mkdir build` (Creates a build directory)
 2. `cd build` (Switches into the build directory)
-3. `cmake ..` (Actually runs cmake)
+3. `cmake .. -Dwarnings-as-errors=OFF` (Actually runs cmake)
```
Comment (Krzmbrzl) [inline docs/dev/build-instructions/build_macos.md:35]: This should not be part of the commit for two reasons:
1. We don't generally want to advise people to disable warnings
2. Even if we wanted to, this would have to be a separate commit of type `DOC`

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]

File: docs/dev/build-instructions/build_macos.md (hunk 1 above)
Comment (Krzmbrzl) [reply docs/dev/build-instructions/build_macos.md:35]: If you want to, you can create an entry in the build-faq that tells people that they can disable warnings-as-errors if they encounter a build failure related to that (in a separate commit)

Comment (saerdnaer) [reply docs/dev/build-instructions/build_macos.md:35]: This does not disable warnings, it just changes their severity from breaking to display.

Comment (Krzmbrzl) [reply docs/dev/build-instructions/build_macos.md:35]: Yes I know - but let's be honest: It will cause people to ignore warnings when doing their development. That's not what we want.

**Comment by saerdnaer** [comment]
obsolted by #5352

**Comment by Krzmbrzl** [comment]
Thank you for your contribution, nonetheless!
